File: solutions/cpp/loggingframework/FileAppender.cpp
#include "FileAppender.hpp"

FileAppender::FileAppender(const std::string& filename) 
    : filename(filename) {
    file.open(filename, std::ios::app);
}

FileAppender::~FileAppender() {
    if (file.is_open()) {
        file.close();
    }
}

void FileAppender::append(const LogMessage& message) {
    if (file.is_open()) {
        file << message.getFormattedMessage() << std::endl;
    }
} 

File: solutions/golang/loggingframework/file_appender.go
package loggingframework

import (
	"os"
)

type FileAppender struct {
	filePath string
}

func NewFileAppender(filePath string) *FileAppender {
	return &FileAppender{
		filePath: filePath,
	}
}

func (a *FileAppender) Append(message *LogMessage) error {
	f, err := os.OpenFile(a.filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		return err
	}
	defer f.Close()

	_, err = f.WriteString(message.String() + "\n")
	return err
}
